subiquity/common/filesystem/gaps.py

```diff
@@ -150,7 +150,7 @@ def maybe_add_gap(start, end, in_extended):
         if part is None:
             gap_end = ad(device.size - info.min_end_offset)
         else:
-            gap_end = ad(part.offset)
+            gap_end = ad(part.offset - info.ebr_space)
```

Comment:
It does not feel right to me to subtract `info.ebr_space` if we're not dealing with a logical partition.
Shouldn't it be:
```python
if part.is_logical():
    gap_end = ad(part.offset - info.ebr_space)
else:
    gap_end = ad(part.offset)
```

Reply:
Excellent spot.  This would have short-changed primary partitions.  Fixed, and a new unit test.